### ISSUE TITLE
Rev 1.1.1 failed to start but 1.0.0 worked

### ISSUE THREAD

**Issue by PhillipOConnor** [mtkennerly/shawl#25]
I installed the windows 64 bit version 1.1.1 on Microsoft Windows Server 2019 Datacenter Version 10.0.17763 Build 17763. The command to create the service was:
`sc create Syncthing DisplayName= "Syncthing" start= delayed-auto depend= NlaSvc binPath= "C:\syncthing\shawl.exe run --no-restart --cwd C:\syncthing --name Syncthing --restart-if 3,4 --stop-timeout 10000 -- C:\syncthing\syncthing.exe -no-restart -no-browser -home=C:\syncthing\home"`
I used the services gui to set the user and password.

The service failed to start and the following (edited) events occurred:
> Log Name:      System
Source:        Service Control Manager
Date:          10/10/2022 7:06:35 PM
Event ID:      7009
Task Category: None
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      servername.local
Description:
A timeout was reached (300000 milliseconds) while waiting for the Syncthing service to connect.

> Log Name:      System
Source:        Service Control Manager
Date:          10/10/2022 7:06:35 PM
Event ID:      7000
Task Category: None
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      servername.local
Description:
The Syncthing service failed to start due to the following error: 
The service did not respond to the start or control request in a timely fashion.

I replaced version 1.1.1 with 1.0.0 and it then worked as expected.

**Comment by mtkennerly** [comment]
Hi! Your command includes both `--no-restart` and `--restart-if 3,4`, which will produce an error on v1.1.1. On older versions, the combination was allowed, but that was a bug. In this case, it would implicitly give priority to `--restart-if` and ignore `--no-restart`.

**Comment by PhillipOConnor** [comment]
Thanks, I will try again with the corrected command. I had two problems resolving this myself.

Firstly, I misinterpreted the documentation.
>  --(no-)restart for all exit codes and --restart-if(-not) for specific exit codes

It was not clear to me that these options were exclusive, maybe "or" would be a better connector or some other clearer statement could be made about their use.

Secondly, I looked for the log file to get a clue to the problem but all I found was the event log entries I included. Is the startup problem logged anywhere?

**Comment by mtkennerly** [comment]
> It was not clear to me that these options were exclusive, maybe "or" would be a better connector or some other clearer statement could be made about their use.

That's a good point. I'll update the note.

Shawl's `add` command gives you a descriptive error, but of course that doesn't help when you use `sc` directly:

```
$ shawl add --name foo --restart --no-restart -- foo
error: The argument '--restart' cannot be used with '--no-restart'
```

> Secondly, I looked for the log file to get a clue to the problem but all I found was the event log entries I included. Is the startup problem logged anywhere?

It should be in `C:\syncthing` along with your copy of shawl.exe (`shawl_*.log`).

**Comment by PhillipOConnor** [comment]
That directory was the first place I looked but there was no log. There is now but the first entry is from when I put in 1.0.0

I assume that the argument problem is detected and acted on before the selection of the working directory and opening the log file. I scanned the full disk contents for shawl*.log but the only copy found was in c:\syncthing with entries after the fault was first addressed. This installation of shawl and syncthing is part of a much larger urgent job due to be completed this Saturday so I could take time to read the source code next week to see what is going on if my Rust skills are up to it. If I can I will try to make some constructive comments.

I switched to shawl from other ways of getting syncthing running constantly and have found it to be the best solution for me so far. I would like to thank you for developing and supporting this application.
